virtool/analyses/data.py

```diff
@@ -1,23 +1,391 @@
+from sqlalchemy.ext.asyncio import AsyncEngine, AsyncSession
+from virtool_core.models.analysis import AnalysisSearchResult, Analysis
+
+import virtool.samples.db
 import virtool.analyses.format
-from virtool.types import Document
+from virtool.analyses.db import PROJECTION, processor
+from virtool.analyses.files import create_analysis_file
+from virtool.analyses.models import AnalysisFile
+from virtool.analyses.utils import attach_analysis_files, find_nuvs_sequence_by_index
+from virtool.api.utils import paginate
+from virtool.blast.models import NuVsBlast
+from virtool.blast.task import BLASTTask
+from virtool.blast.transform import AttachNuVsBLAST
+from virtool.data.errors import (
+    ResourceNotFoundError,
+    ResourceNotModifiedError,
+    ResourceError,
+    ResourceConflictError,
+)
+from virtool.mongo.core import DB
+import virtool.uploads.db
+from virtool.mongo.transforms import apply_transforms
+from virtool.mongo.utils import get_one_field
+from virtool.pg.utils import delete_row, get_row_by_id
+from virtool.samples.utils import get_sample_rights
+from virtool.subtractions.db import AttachSubtractionTransform
+from virtool.tasks.client import TasksClient
+from virtool.uploads.utils import naive_writer
+
+from virtool.users.db import AttachUserTransform
+from virtool.utils import run_in_thread, rm
+from virtool.samples.db import recalculate_workflow_tags
+
+
+logger = getLogger("analyses")
 
 
 class AnalysisData:
-    def __init__(self, db):
+    def __init__(self, db: DB, config, pg: AsyncEngine, tasks: TasksClient):
         self._db = db
+        self._config = config
+        self._pg = pg
+        self._tasks = tasks
+
+    async def find(
+        self, query: Union[Dict, MultiDictProxy[str]], client
+    ) -> AnalysisSearchResult:
+        """
+        List all analysis documents.
+
+        :param query: the request query
+        :param client: the client object
+        :return: a list of all analysis documents
+        """
+        db_query = {}
+
+        data = await paginate(
+            self._db.analyses,
+            db_query,
+            query,
+            projection=PROJECTION,
+            sort=[("created_at", -1)],
+        )
+
+        per_document_can_read = await gather(
+            *[
+                virtool.samples.db.check_rights(
+                    self._db, document["sample"]["id"], client, write=False
+                )
+                for document in data["documents"]
+            ]
+        )
+
+        documents = [
+            document
+            for document, can_write in zip(data["documents"], per_document_can_read)
+            if can_write
+        ]
+
+        documents = await apply_transforms(
+            documents,
+            [AttachUserTransform(self._db), AttachSubtractionTransform(self._db)],
+        )
+
+        data["documents"] = documents
 
-    async def get_by_id(self, analysis_id: str) -> Optional[Document]:
+        return AnalysisSearchResult(**data)
```

Comment:
I think this will work:
```py
return AnalysisSearchResult(**data, document=documents)
```

Reply:
I tried it and it wasnt working so just left it as is.